<!DOCTYPE html>
<html lang="en"><head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1"><!-- Begin Jekyll SEO tag v2.5.0 -->
<title>Bioinformatics tools (biology and statistics description) | Journey of a Bioinformatician</title>
<meta name="generator" content="Jekyll v3.8.3" />
<meta property="og:title" content="Bioinformatics tools (biology and statistics description)" />
<meta name="author" content="Aarthi Ramakrishnan" />
<meta property="og:locale" content="en_US" />
<meta name="description" content="Tools for quality control and adapter trimming -&nbsp;FastQC - Checking the quality of raw reads, requires little memory,cutadapt - Trimming the adapter sequences, supports wider range of sequencing platforms compared to fastx toolkitFastx toolkit - Checking read quality, trimming adapter sequencesRNA-Seqc - RNA-SeQC is used after FASTQC, which allows users to investigate the yield, alignment and duplication rate, GC bias, rRNA contamination rate, exon vs intron region vs intragenic alignment, 3’5’ bias, coverage, computationally intensivengsplot - Finding overall coverage of raw RNA-seq reads from transcription start site to transcription end siteTrimGalore - wrapper of cutadapt and fastqcTools for alignment -&nbsp;Bowtie - Aligner, not splice awareBowtie2 - Aligner, not splice awareTophat - Aligner, splice aware, successor of bowtie (wraps around bowtie), splice detectionTophat2 - Aligner, splice awareHISAT (Hierarchical indexing for spliced alignment of Transcripts) - A fast aligner, splice aware, specific to RNA, does NOT do soft clippingHISAT2 - A fast aligner, splice aware, specific to RNA, Soft clipping doneSTAR (Spliced Transcripts Alignment to a Reference) - A fast aligner (similar to HISAT), splice awareBWA (Burrow wheeler aligner) - DNA alignerTrinity - De novo transcriptome assemblyBSMAP - tolerate conversion of c to t in bisulfite sequencingBismark - tolerate conversion of c to t in bisulfite sequencingCufflinks - denovo transcriptome assemblyMisc -SAMtools - conversion of sam to bam, bam to sorted bam, removal of duplicate readsBedtools - bedintersect,&nbsp;Cuffmergepicard - removal of duplicate reads, merging sam filesdiffRepeats - quantify repeat element enrichment / differential analysis of repeat elements in the DNA, algning short reads to a database of repeat sequencesdiffReps - differential enrichment of a histone mark/transcription factormirdeep - de novo prediction of miRNA from RNA-seq datangsplot - can accept large files, runs with small memory footprintsphantompeek -chippeekanno -region_analysis -&nbsp;GeneOverlap - used to find the overlap between 2 sets of genesIdentification of splicing events -&nbsp;rMATS - exon skipping, intron retentionMISO - Mixture of isoforms, intron retention, skipped exon, mutually exclusive and alternative 3&#39; UTR, splicing ratio b/w isoforms of same gene, splicing ratio between same isoform from 2 different conditions, uses bayesian theorem to iteratively infer splicing ratio. Allows to infer splicing ratios, both among isoforms of the same gene and between 2 conditions for the same isoform. Helps to identify contribution of histone/tf in regulation of AS.Peak calling -DANPOSMACSHomerQuantification of aligned reads -Cufflinks - Transcriptome abundance, de-novo transcriptome assembly, based on maximum likelihood principle, groups transcripts with the same transcription start siteStringTie - Transcriptome abundance, exon level quantificationSailfish - no need of aligner, direct quantificationDifferential expression analysis -&nbsp;Cuffdiff - following cufflinks, generates high false positivesCuffdiff2 - generates high false positivesedgeR - Based on bayesian methods, negative binomial testingDEseq - Based on bayesian methods, local regression, generates high false negatives (highly conservative output)Limma -&nbsp;Voom - models the mean variance relationship on log transformed read countshtseq countDifferential chip-seq analysis -diffReps - to compare differential enrichment of a histone mark/transcription factor b/w 2 conditions, takes into account biological variations between a group of samplesRemove batch effects from RNA-seq datasets -&nbsp;ComBatBatchQCBioinformatics databases -repbase - documents repetitive DNA sequences across eukaryotic speciesmiRbase -piRNAbank -lncRNAdb-snornabase -SILVA -Rfam -NONCODE -fRNAdb -" />
<meta property="og:description" content="Tools for quality control and adapter trimming -&nbsp;FastQC - Checking the quality of raw reads, requires little memory,cutadapt - Trimming the adapter sequences, supports wider range of sequencing platforms compared to fastx toolkitFastx toolkit - Checking read quality, trimming adapter sequencesRNA-Seqc - RNA-SeQC is used after FASTQC, which allows users to investigate the yield, alignment and duplication rate, GC bias, rRNA contamination rate, exon vs intron region vs intragenic alignment, 3’5’ bias, coverage, computationally intensivengsplot - Finding overall coverage of raw RNA-seq reads from transcription start site to transcription end siteTrimGalore - wrapper of cutadapt and fastqcTools for alignment -&nbsp;Bowtie - Aligner, not splice awareBowtie2 - Aligner, not splice awareTophat - Aligner, splice aware, successor of bowtie (wraps around bowtie), splice detectionTophat2 - Aligner, splice awareHISAT (Hierarchical indexing for spliced alignment of Transcripts) - A fast aligner, splice aware, specific to RNA, does NOT do soft clippingHISAT2 - A fast aligner, splice aware, specific to RNA, Soft clipping doneSTAR (Spliced Transcripts Alignment to a Reference) - A fast aligner (similar to HISAT), splice awareBWA (Burrow wheeler aligner) - DNA alignerTrinity - De novo transcriptome assemblyBSMAP - tolerate conversion of c to t in bisulfite sequencingBismark - tolerate conversion of c to t in bisulfite sequencingCufflinks - denovo transcriptome assemblyMisc -SAMtools - conversion of sam to bam, bam to sorted bam, removal of duplicate readsBedtools - bedintersect,&nbsp;Cuffmergepicard - removal of duplicate reads, merging sam filesdiffRepeats - quantify repeat element enrichment / differential analysis of repeat elements in the DNA, algning short reads to a database of repeat sequencesdiffReps - differential enrichment of a histone mark/transcription factormirdeep - de novo prediction of miRNA from RNA-seq datangsplot - can accept large files, runs with small memory footprintsphantompeek -chippeekanno -region_analysis -&nbsp;GeneOverlap - used to find the overlap between 2 sets of genesIdentification of splicing events -&nbsp;rMATS - exon skipping, intron retentionMISO - Mixture of isoforms, intron retention, skipped exon, mutually exclusive and alternative 3&#39; UTR, splicing ratio b/w isoforms of same gene, splicing ratio between same isoform from 2 different conditions, uses bayesian theorem to iteratively infer splicing ratio. Allows to infer splicing ratios, both among isoforms of the same gene and between 2 conditions for the same isoform. Helps to identify contribution of histone/tf in regulation of AS.Peak calling -DANPOSMACSHomerQuantification of aligned reads -Cufflinks - Transcriptome abundance, de-novo transcriptome assembly, based on maximum likelihood principle, groups transcripts with the same transcription start siteStringTie - Transcriptome abundance, exon level quantificationSailfish - no need of aligner, direct quantificationDifferential expression analysis -&nbsp;Cuffdiff - following cufflinks, generates high false positivesCuffdiff2 - generates high false positivesedgeR - Based on bayesian methods, negative binomial testingDEseq - Based on bayesian methods, local regression, generates high false negatives (highly conservative output)Limma -&nbsp;Voom - models the mean variance relationship on log transformed read countshtseq countDifferential chip-seq analysis -diffReps - to compare differential enrichment of a histone mark/transcription factor b/w 2 conditions, takes into account biological variations between a group of samplesRemove batch effects from RNA-seq datasets -&nbsp;ComBatBatchQCBioinformatics databases -repbase - documents repetitive DNA sequences across eukaryotic speciesmiRbase -piRNAbank -lncRNAdb-snornabase -SILVA -Rfam -NONCODE -fRNAdb -" />
<link rel="canonical" href="http://localhost:4000/2017/12/24/bioinformatics-tools-biology-and.html" />
<meta property="og:url" content="http://localhost:4000/2017/12/24/bioinformatics-tools-biology-and.html" />
<meta property="og:site_name" content="Journey of a Bioinformatician" />
<meta property="og:type" content="article" />
<meta property="article:published_time" content="2017-12-24T03:50:00-05:00" />
<script type="application/ld+json">
{"url":"http://localhost:4000/2017/12/24/bioinformatics-tools-biology-and.html","mainEntityOfPage":{"@type":"WebPage","@id":"http://localhost:4000/2017/12/24/bioinformatics-tools-biology-and.html"},"author":{"@type":"Person","name":"Aarthi Ramakrishnan"},"description":"Tools for quality control and adapter trimming -&nbsp;FastQC - Checking the quality of raw reads, requires little memory,cutadapt - Trimming the adapter sequences, supports wider range of sequencing platforms compared to fastx toolkitFastx toolkit - Checking read quality, trimming adapter sequencesRNA-Seqc - RNA-SeQC is used after FASTQC, which allows users to investigate the yield, alignment and duplication rate, GC bias, rRNA contamination rate, exon vs intron region vs intragenic alignment, 3’5’ bias, coverage, computationally intensivengsplot - Finding overall coverage of raw RNA-seq reads from transcription start site to transcription end siteTrimGalore - wrapper of cutadapt and fastqcTools for alignment -&nbsp;Bowtie - Aligner, not splice awareBowtie2 - Aligner, not splice awareTophat - Aligner, splice aware, successor of bowtie (wraps around bowtie), splice detectionTophat2 - Aligner, splice awareHISAT (Hierarchical indexing for spliced alignment of Transcripts) - A fast aligner, splice aware, specific to RNA, does NOT do soft clippingHISAT2 - A fast aligner, splice aware, specific to RNA, Soft clipping doneSTAR (Spliced Transcripts Alignment to a Reference) - A fast aligner (similar to HISAT), splice awareBWA (Burrow wheeler aligner) - DNA alignerTrinity - De novo transcriptome assemblyBSMAP - tolerate conversion of c to t in bisulfite sequencingBismark - tolerate conversion of c to t in bisulfite sequencingCufflinks - denovo transcriptome assemblyMisc -SAMtools - conversion of sam to bam, bam to sorted bam, removal of duplicate readsBedtools - bedintersect,&nbsp;Cuffmergepicard - removal of duplicate reads, merging sam filesdiffRepeats - quantify repeat element enrichment / differential analysis of repeat elements in the DNA, algning short reads to a database of repeat sequencesdiffReps - differential enrichment of a histone mark/transcription factormirdeep - de novo prediction of miRNA from RNA-seq datangsplot - can accept large files, runs with small memory footprintsphantompeek -chippeekanno -region_analysis -&nbsp;GeneOverlap - used to find the overlap between 2 sets of genesIdentification of splicing events -&nbsp;rMATS - exon skipping, intron retentionMISO - Mixture of isoforms, intron retention, skipped exon, mutually exclusive and alternative 3&#39; UTR, splicing ratio b/w isoforms of same gene, splicing ratio between same isoform from 2 different conditions, uses bayesian theorem to iteratively infer splicing ratio. Allows to infer splicing ratios, both among isoforms of the same gene and between 2 conditions for the same isoform. Helps to identify contribution of histone/tf in regulation of AS.Peak calling -DANPOSMACSHomerQuantification of aligned reads -Cufflinks - Transcriptome abundance, de-novo transcriptome assembly, based on maximum likelihood principle, groups transcripts with the same transcription start siteStringTie - Transcriptome abundance, exon level quantificationSailfish - no need of aligner, direct quantificationDifferential expression analysis -&nbsp;Cuffdiff - following cufflinks, generates high false positivesCuffdiff2 - generates high false positivesedgeR - Based on bayesian methods, negative binomial testingDEseq - Based on bayesian methods, local regression, generates high false negatives (highly conservative output)Limma -&nbsp;Voom - models the mean variance relationship on log transformed read countshtseq countDifferential chip-seq analysis -diffReps - to compare differential enrichment of a histone mark/transcription factor b/w 2 conditions, takes into account biological variations between a group of samplesRemove batch effects from RNA-seq datasets -&nbsp;ComBatBatchQCBioinformatics databases -repbase - documents repetitive DNA sequences across eukaryotic speciesmiRbase -piRNAbank -lncRNAdb-snornabase -SILVA -Rfam -NONCODE -fRNAdb -","@type":"BlogPosting","headline":"Bioinformatics tools (biology and statistics description)","dateModified":"2017-12-24T03:50:00-05:00","datePublished":"2017-12-24T03:50:00-05:00","@context":"http://schema.org"}</script>
<!-- End Jekyll SEO tag -->
<link rel="stylesheet" href="/assets/main.css"><link type="application/atom+xml" rel="alternate" href="http://localhost:4000/feed.xml" title="Journey of a Bioinformatician" /></head>
<body><header class="site-header" role="banner">

  <div class="wrapper"><a class="site-title" rel="author" href="/">Journey of a Bioinformatician</a><nav class="site-nav">
        <input type="checkbox" id="nav-trigger" class="nav-trigger" />
        <label for="nav-trigger">
          <span class="menu-icon">
            <svg viewBox="0 0 18 15" width="18px" height="15px">
              <path d="M18,1.484c0,0.82-0.665,1.484-1.484,1.484H1.484C0.665,2.969,0,2.304,0,1.484l0,0C0,0.665,0.665,0,1.484,0 h15.032C17.335,0,18,0.665,18,1.484L18,1.484z M18,7.516C18,8.335,17.335,9,16.516,9H1.484C0.665,9,0,8.335,0,7.516l0,0 c0-0.82,0.665-1.484,1.484-1.484h15.032C17.335,6.031,18,6.696,18,7.516L18,7.516z M18,13.516C18,14.335,17.335,15,16.516,15H1.484 C0.665,15,0,14.335,0,13.516l0,0c0-0.82,0.665-1.483,1.484-1.483h15.032C17.335,12.031,18,12.695,18,13.516L18,13.516z"/>
            </svg>
          </span>
        </label>

        <div class="trigger"><a class="page-link" href="/about/">About</a></div>
      </nav></div>
</header>
<main class="page-content" aria-label="Content">
      <div class="wrapper">
        <article class="post h-entry" itemscope itemtype="http://schema.org/BlogPosting">

  <header class="post-header">
    <h1 class="post-title p-name" itemprop="name headline">Bioinformatics tools (biology and statistics description)</h1>
    <p class="post-meta">
      <time class="dt-published" datetime="2017-12-24T03:50:00-05:00" itemprop="datePublished">Dec 24, 2017
      </time>• <span itemprop="author" itemscope itemtype="http://schema.org/Person"><span class="p-author h-card" itemprop="name">Aarthi Ramakrishnan</span></span></p>
  </header>

  <div class="post-content e-content" itemprop="articleBody">
    <div dir="ltr" style="text-align: left;" trbidi="on"><span style="font-family: inherit;"><b>Tools for quality control and adapter trimming -&nbsp;</b></span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">FastQC - Checking the quality of raw reads, requires little memory,</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">cutadapt - Trimming the adapter sequences, supports wider range of sequencing platforms compared to fastx toolkit</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">Fastx toolkit - Checking read quality, trimming adapter sequences</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">RNA-Seqc - RNA-SeQC is used after FASTQC, which allows users to investigate the yield, alignment and duplication rate, GC bias, rRNA contamination rate, exon vs intron region vs intragenic alignment, 3’5’ bias, coverage, computationally intensive</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">ngsplot - Finding overall coverage of raw RNA-seq reads from transcription start site to transcription end site</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">TrimGalore - wrapper of cutadapt and fastqc</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><b>Tools for alignment -&nbsp;</b></span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">Bowtie - Aligner, not splice aware</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">Bowtie2 - Aligner, not splice aware</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">Tophat - Aligner, splice aware, successor of bowtie (wraps around bowtie), splice detection</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">Tophat2 - Aligner, splice aware</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">HISAT (Hierarchical indexing for spliced alignment of Transcripts) - A fast aligner, splice aware, specific to RNA, does NOT do soft clipping</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">HISAT2 - A fast aligner, splice aware, specific to RNA, Soft clipping done</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">STAR (Spliced Transcripts Alignment to a Reference) - A fast aligner (similar to HISAT), splice aware</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">BWA (Burrow wheeler aligner) - DNA aligner</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">Trinity - De novo transcriptome assembly</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">BSMAP - tolerate conversion of c to t in bisulfite sequencing</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">Bismark - tolerate conversion of c to t in bisulfite sequencing</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">Cufflinks - denovo transcriptome assembly</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><b>Misc -</b></span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">SAMtools - conversion of sam to bam, bam to sorted bam, removal of duplicate reads</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">Bedtools - bedintersect,&nbsp;</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">Cuffmerge</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">picard - removal of duplicate reads, merging sam files</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">diffRepeats - quantify repeat element enrichment / differential analysis of repeat elements in the DNA, algning short reads to a database of repeat sequences</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">diffReps - differential enrichment of a histone mark/transcription factor</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">mirdeep - de novo prediction of miRNA from RNA-seq data</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">ngsplot - can accept large files, runs with small memory footprints</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">phantompeek -</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">chippeekanno -</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">region_analysis -&nbsp;</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">GeneOverlap - used to find the overlap between 2 sets of genes</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><b>Identification of splicing events -&nbsp;</b></span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">rMATS - exon skipping, intron retention</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">MISO - Mixture of isoforms, intron retention, skipped exon, mutually exclusive and alternative 3' UTR, splicing ratio b/w isoforms of same gene, splicing ratio between same isoform from 2 different conditions, uses bayesian theorem to iteratively infer splicing ratio. Allows to infer splicing ratios, both among isoforms of the same gene and between 2 conditions for the same isoform. Helps to identify contribution of histone/tf in regulation of AS.</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><b>Peak calling -</b></span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">DANPOS</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">MACS</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">Homer</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><b>Quantification of aligned reads -</b></span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">Cufflinks - Transcriptome abundance, de-novo transcriptome assembly, based on maximum likelihood principle, groups transcripts with the same transcription start site</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">StringTie - Transcriptome abundance, exon level quantification</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">Sailfish - no need of aligner, direct quantification</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><b>Differential expression analysis -&nbsp;</b></span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">Cuffdiff - following cufflinks, generates high false positives</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">Cuffdiff2 - generates high false positives</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">edgeR - Based on bayesian methods, negative binomial testing</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">DEseq - Based on bayesian methods, local regression, generates high false negatives (highly conservative output)</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">Limma -&nbsp;</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">Voom - models the mean variance relationship on log transformed read counts</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">htseq count</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><b>Differential chip-seq analysis -</b></span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">diffReps - to compare differential enrichment of a histone mark/transcription factor b/w 2 conditions, takes into account biological variations between a group of samples</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><b>Remove batch effects from RNA-seq datasets -&nbsp;</b></span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">ComBat</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">BatchQC</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><b>Bioinformatics databases -</b></span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">repbase - documents repetitive DNA sequences across eukaryotic species</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">miRbase -</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">piRNAbank -</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">lncRNAdb-</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">snornabase -</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">SILVA -</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">Rfam -</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">NONCODE -</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">fRNAdb -</span></div>
  </div><a class="u-url" href="/2017/12/24/bioinformatics-tools-biology-and.html" hidden></a>
</article>

      </div>
    </main><footer class="site-footer h-card">
  <data class="u-url" href="/"></data>

  <div class="wrapper">

    <h2 class="footer-heading">Journey of a Bioinformatician</h2>

    <div class="footer-col-wrapper">
      <div class="footer-col footer-col-1">
        <ul class="contact-list">
          <li class="p-name">Journey of a Bioinformatician</li><li><a class="u-email" href="mailto:aarthir239@gmail.com">aarthir239@gmail.com</a></li></ul>
      </div>

      <div class="footer-col footer-col-2"><ul class="social-media-list"><li><a href="https://github.com/aartrama"><svg class="svg-icon"><use xlink:href="/assets/minima-social-icons.svg#github"></use></svg> <span class="username">aartrama</span></a></li></ul>
</div>

      <div class="footer-col footer-col-3">
        <p>Aarthi Ramakrishnan&#39;s blog</p>
      </div>
    </div>

  </div>

</footer>
</body>

</html>
