---
layout: post
title: Bioinformatics terms
date: '2017-12-24T00:32:00.001-08:00'
author: Aarthi Ramakrishnan
tags: 
modified_time: '2018-05-12T14:38:44.567-07:00'
blogger_id: tag:blogger.com,1999:blog-2313420884919503029.post-3420105469626882564
blogger_orig_url: http://aarthiramakrishnan.blogspot.com/2017/12/bioinformatics-terms.html
---

<div dir="ltr" style="text-align: left;" trbidi="on">Following are some of the common bioinformatics terms one comes across in technical papers:<br /><br />1) <b>Coverage</b> - total number of mapped reads divided by the size of the reference genome (<a href="https://www.biostars.org/p/5165">https://www.biostars.org/p/5165</a>)<br /><br /><table>    <tbody><tr>        <td>22</td>        <td>&nbsp; 2</td>        <td>&nbsp; 4</td>        <td>&nbsp; 137928</td>        <td>&nbsp; 2.9e-05</td>    </tr><tr>        <td>22</td>        <td>&nbsp; 3</td>        <td>&nbsp; 28</td>        <td>&nbsp; 137928</td>        <td>&nbsp; 6.4e-05</td>    </tr><tr>        <td>22</td>        <td>&nbsp; 4</td>        <td>&nbsp; 10</td>        <td>&nbsp; 137928</td>        <td>&nbsp; 7.25e-05</td>    </tr></tbody></table><br />The above means that 4 bases (in that contig/chromosome) (out of 137928 bases) have a coverage of 2 (or have been covered 2 times), 28 bases have a coverage of 3, 10 bases have a coverage of 4 and so on. <br />If you want to get the average coverage: add up the product of bases per coverage [2* 4+3* 28+4* 10+...] and divide by the total number of bases [137928].<br /><br /><br />2) <b>Contigs</b> - contigs include completed chromosomes as well as incomplete chromosomes<br /><br />3) <b>Demultiplex</b> - differentiate reads from each sample<br /><br />4) <b>Barcode/adapter sequence</b> - the sequences used to identify a sample<br /><br />Note - adapter removal does NOT drastically improve the differential expression analysis results.<br /><br /><br />5) <b>Soft and hard clipped bases</b> - <br /><br /><u>Soft-clipped:</u> bases in 5' and 3' of the read are NOT part of the alignment.<br /><br /><u>Hard-clipped:</u> bases in 5' and 3' of the read are NOT part of the alignment AND those bases have been removed from the read sequence in the BAM file. The 'real' sequence length would be length of sequence + count-of-hard-clipped-bases<br /><br />6) <b>Principal Component Analysis</b> -<br /><br />7) <b>Sequencing depth</b> - The number of times a particular base has been sequenced.</div>