---
layout: post
title: Bioinformatics tools (biology and statistics description)
date: '2017-12-24T00:50:00.000-08:00'
author: Aarthi Ramakrishnan
tags: 
modified_time: '2018-05-30T09:56:38.203-07:00'
blogger_id: tag:blogger.com,1999:blog-2313420884919503029.post-5428854578249443505
blogger_orig_url: http://aarthiramakrishnan.blogspot.com/2017/12/bioinformatics-tools-biology-and.html
---

<div dir="ltr" style="text-align: left;" trbidi="on"><span style="font-family: inherit;"><b>Tools for quality control and adapter trimming -&nbsp;</b></span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">FastQC - Checking the quality of raw reads, requires little memory,</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">cutadapt - Trimming the adapter sequences, supports wider range of sequencing platforms compared to fastx toolkit</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">Fastx toolkit - Checking read quality, trimming adapter sequences</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">RNA-Seqc - RNA-SeQC is used after FASTQC, which allows users to investigate the yield, alignment and duplication rate, GC bias, rRNA contamination rate, exon vs intron region vs intragenic alignment, 3’5’ bias, coverage, computationally intensive</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">ngsplot - Finding overall coverage of raw RNA-seq reads from transcription start site to transcription end site</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">TrimGalore - wrapper of cutadapt and fastqc</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><b>Tools for alignment -&nbsp;</b></span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">Bowtie - Aligner, not splice aware</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">Bowtie2 - Aligner, not splice aware</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">Tophat - Aligner, splice aware, successor of bowtie (wraps around bowtie), splice detection</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">Tophat2 - Aligner, splice aware</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">HISAT (Hierarchical indexing for spliced alignment of Transcripts) - A fast aligner, splice aware, specific to RNA, does NOT do soft clipping</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">HISAT2 - A fast aligner, splice aware, specific to RNA, Soft clipping done</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">STAR (Spliced Transcripts Alignment to a Reference) - A fast aligner (similar to HISAT), splice aware</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">BWA (Burrow wheeler aligner) - DNA aligner</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">Trinity - De novo transcriptome assembly</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">BSMAP - tolerate conversion of c to t in bisulfite sequencing</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">Bismark - tolerate conversion of c to t in bisulfite sequencing</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">Cufflinks - denovo transcriptome assembly</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><b>Misc -</b></span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">SAMtools - conversion of sam to bam, bam to sorted bam, removal of duplicate reads</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">Bedtools - bedintersect,&nbsp;</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">Cuffmerge</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">picard - removal of duplicate reads, merging sam files</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">diffRepeats - quantify repeat element enrichment / differential analysis of repeat elements in the DNA, algning short reads to a database of repeat sequences</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">diffReps - differential enrichment of a histone mark/transcription factor</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">mirdeep - de novo prediction of miRNA from RNA-seq data</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">ngsplot - can accept large files, runs with small memory footprints</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">phantompeek -</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">chippeekanno -</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">region_analysis -&nbsp;</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">GeneOverlap - used to find the overlap between 2 sets of genes</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><b>Identification of splicing events -&nbsp;</b></span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">rMATS - exon skipping, intron retention</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">MISO - Mixture of isoforms, intron retention, skipped exon, mutually exclusive and alternative 3' UTR, splicing ratio b/w isoforms of same gene, splicing ratio between same isoform from 2 different conditions, uses bayesian theorem to iteratively infer splicing ratio. Allows to infer splicing ratios, both among isoforms of the same gene and between 2 conditions for the same isoform. Helps to identify contribution of histone/tf in regulation of AS.</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><b>Peak calling -</b></span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">DANPOS</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">MACS</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">Homer</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><b>Quantification of aligned reads -</b></span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">Cufflinks - Transcriptome abundance, de-novo transcriptome assembly, based on maximum likelihood principle, groups transcripts with the same transcription start site</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">StringTie - Transcriptome abundance, exon level quantification</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">Sailfish - no need of aligner, direct quantification</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><b>Differential expression analysis -&nbsp;</b></span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">Cuffdiff - following cufflinks, generates high false positives</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">Cuffdiff2 - generates high false positives</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">edgeR - Based on bayesian methods, negative binomial testing</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">DEseq - Based on bayesian methods, local regression, generates high false negatives (highly conservative output)</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">Limma -&nbsp;</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">Voom - models the mean variance relationship on log transformed read counts</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">htseq count</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><b>Differential chip-seq analysis -</b></span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">diffReps - to compare differential enrichment of a histone mark/transcription factor b/w 2 conditions, takes into account biological variations between a group of samples</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><b>Remove batch effects from RNA-seq datasets -&nbsp;</b></span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">ComBat</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">BatchQC</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><b>Bioinformatics databases -</b></span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">repbase - documents repetitive DNA sequences across eukaryotic species</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">miRbase -</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">piRNAbank -</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">lncRNAdb-</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">snornabase -</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">SILVA -</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">Rfam -</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">NONCODE -</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">fRNAdb -</span></div>