<!DOCTYPE html>
<html lang="en"><head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1"><!-- Begin Jekyll SEO tag v2.5.0 -->
<title>stringsAsFactors trouble in R | Journey of a Bioinformatician</title>
<meta name="generator" content="Jekyll v3.8.3" />
<meta property="og:title" content="stringsAsFactors trouble in R" />
<meta name="author" content="Aarthi Ramakrishnan" />
<meta property="og:locale" content="en_US" />
<meta name="description" content="Recently, I was reading a counts matrix into R -&gt; counts &lt;- read.table(&quot;htseq_counts_matrix.txt&quot;)&gt; head(counts) Pool5_RNAseq_12_S33_L006 Pool5_RNAseq_19_S34_L006ENSMUSG00000000001 4676 2314ENSMUSG00000000003 0 0ENSMUSG00000000028 558 172ENSMUSG00000000031 25272 6334ENSMUSG00000000037 28 118ENSMUSG00000000049 3 0 Pool5_RNAseq_4_S29_L006 Pool5_RNAseq_5_S30_L006ENSMUSG00000000001 3422 4300ENSMUSG00000000003 0 0ENSMUSG00000000028 328 581ENSMUSG00000000031 7878 22343ENSMUSG00000000037 212 21ENSMUSG00000000049 2 3 Pool5_RNAseq_6_S31_L006 Pool5_RNAseq_7_S32_L006ENSMUSG00000000001 2221 857ENSMUSG00000000003 0 0ENSMUSG00000000028 158 159ENSMUSG00000000031 27124 3655ENSMUSG00000000037 5 32ENSMUSG00000000049 0 0I required the counts for 2 gene IDs only. The 2 IDs were stored in a separate file called &#39;some_genes.txt&#39;. &gt; some_genes &lt;- read.table(&quot;some_genes.txt&quot;)&gt; some_genes &lt;- some_genes$V1&gt; some_genes[1] ENSMUSG00000043587 ENSMUSG00000040276Levels: ENSMUSG00000040276 ENSMUSG00000043587The 2 IDs were ENSMUSG00000043587 and ENSMUSG00000040276. When I tried to subset the counts matrix to retain the counts of just these IDs, something unexpected happened.&gt; some_genes_counts &lt;- counts[some_genes, ]&gt; some_genes_counts Pool5_RNAseq_12_S33_L006 Pool5_RNAseq_19_S34_L006ENSMUSG00000000003 0 0ENSMUSG00000000001 4676 2314 Pool5_RNAseq_4_S29_L006 Pool5_RNAseq_5_S30_L006ENSMUSG00000000003 0 0ENSMUSG00000000001 3422 4300 Pool5_RNAseq_6_S31_L006 Pool5_RNAseq_7_S32_L006ENSMUSG00000000003 0 0ENSMUSG00000000001 2221 857 Instead of obtaining the counts for the 2 IDs, I obtained the counts of 2 totally random genes! When I investigated further -&gt; typeof(some_genes)[1] &quot;integer&quot;The variables in &#39;some_genes&#39; were being converted to integers, and the integers were being used as the row index to subset the counts matrix. This is totally undesirable! But once I included &#39;stringsAsFactors = F&#39; while reading the file into R, the problem was fixed.&nbsp;&gt; some_genes &lt;- read.table(&quot;some_genes.txt&quot;, stringsAsFactors = F)&gt; some_genes &lt;- some_genes$V1&gt; some_genes[1] &quot;ENSMUSG00000043587&quot; &quot;ENSMUSG00000040276&quot;&gt; some_genes_counts &lt;- counts[some_genes, ]&gt; some_genes_counts Pool5_RNAseq_12_S33_L006 Pool5_RNAseq_19_S34_L006ENSMUSG00000043587 385 510ENSMUSG00000040276 5 942 Pool5_RNAseq_4_S29_L006 Pool5_RNAseq_5_S30_L006ENSMUSG00000043587 836 365ENSMUSG00000040276 1021 4 Pool5_RNAseq_6_S31_L006 Pool5_RNAseq_7_S32_L006ENSMUSG00000043587 44 229ENSMUSG00000040276 12 515&gt; typeof(some_genes)[1] &quot;character&quot;This time, I obtained the counts of the right set of IDs. R by default sets stringsAsFactors = T because this setting is convenient while running models using the glm() or lm() function.&nbsp;An alternative to including stringsAsFactors = F for every file you read into R is to just set -&gt; options(stringsAsFactors = FALSE)at the beginning of the R scriptThe following article was also very helpful in getting a better understanding of this problem:https://simplystatistics.org/2015/07/24/stringsasfactors-an-unauthorized-biography/" />
<meta property="og:description" content="Recently, I was reading a counts matrix into R -&gt; counts &lt;- read.table(&quot;htseq_counts_matrix.txt&quot;)&gt; head(counts) Pool5_RNAseq_12_S33_L006 Pool5_RNAseq_19_S34_L006ENSMUSG00000000001 4676 2314ENSMUSG00000000003 0 0ENSMUSG00000000028 558 172ENSMUSG00000000031 25272 6334ENSMUSG00000000037 28 118ENSMUSG00000000049 3 0 Pool5_RNAseq_4_S29_L006 Pool5_RNAseq_5_S30_L006ENSMUSG00000000001 3422 4300ENSMUSG00000000003 0 0ENSMUSG00000000028 328 581ENSMUSG00000000031 7878 22343ENSMUSG00000000037 212 21ENSMUSG00000000049 2 3 Pool5_RNAseq_6_S31_L006 Pool5_RNAseq_7_S32_L006ENSMUSG00000000001 2221 857ENSMUSG00000000003 0 0ENSMUSG00000000028 158 159ENSMUSG00000000031 27124 3655ENSMUSG00000000037 5 32ENSMUSG00000000049 0 0I required the counts for 2 gene IDs only. The 2 IDs were stored in a separate file called &#39;some_genes.txt&#39;. &gt; some_genes &lt;- read.table(&quot;some_genes.txt&quot;)&gt; some_genes &lt;- some_genes$V1&gt; some_genes[1] ENSMUSG00000043587 ENSMUSG00000040276Levels: ENSMUSG00000040276 ENSMUSG00000043587The 2 IDs were ENSMUSG00000043587 and ENSMUSG00000040276. When I tried to subset the counts matrix to retain the counts of just these IDs, something unexpected happened.&gt; some_genes_counts &lt;- counts[some_genes, ]&gt; some_genes_counts Pool5_RNAseq_12_S33_L006 Pool5_RNAseq_19_S34_L006ENSMUSG00000000003 0 0ENSMUSG00000000001 4676 2314 Pool5_RNAseq_4_S29_L006 Pool5_RNAseq_5_S30_L006ENSMUSG00000000003 0 0ENSMUSG00000000001 3422 4300 Pool5_RNAseq_6_S31_L006 Pool5_RNAseq_7_S32_L006ENSMUSG00000000003 0 0ENSMUSG00000000001 2221 857 Instead of obtaining the counts for the 2 IDs, I obtained the counts of 2 totally random genes! When I investigated further -&gt; typeof(some_genes)[1] &quot;integer&quot;The variables in &#39;some_genes&#39; were being converted to integers, and the integers were being used as the row index to subset the counts matrix. This is totally undesirable! But once I included &#39;stringsAsFactors = F&#39; while reading the file into R, the problem was fixed.&nbsp;&gt; some_genes &lt;- read.table(&quot;some_genes.txt&quot;, stringsAsFactors = F)&gt; some_genes &lt;- some_genes$V1&gt; some_genes[1] &quot;ENSMUSG00000043587&quot; &quot;ENSMUSG00000040276&quot;&gt; some_genes_counts &lt;- counts[some_genes, ]&gt; some_genes_counts Pool5_RNAseq_12_S33_L006 Pool5_RNAseq_19_S34_L006ENSMUSG00000043587 385 510ENSMUSG00000040276 5 942 Pool5_RNAseq_4_S29_L006 Pool5_RNAseq_5_S30_L006ENSMUSG00000043587 836 365ENSMUSG00000040276 1021 4 Pool5_RNAseq_6_S31_L006 Pool5_RNAseq_7_S32_L006ENSMUSG00000043587 44 229ENSMUSG00000040276 12 515&gt; typeof(some_genes)[1] &quot;character&quot;This time, I obtained the counts of the right set of IDs. R by default sets stringsAsFactors = T because this setting is convenient while running models using the glm() or lm() function.&nbsp;An alternative to including stringsAsFactors = F for every file you read into R is to just set -&gt; options(stringsAsFactors = FALSE)at the beginning of the R scriptThe following article was also very helpful in getting a better understanding of this problem:https://simplystatistics.org/2015/07/24/stringsasfactors-an-unauthorized-biography/" />
<link rel="canonical" href="http://localhost:4000/2018/06/02/stringsasfactors-trouble-in-r.html" />
<meta property="og:url" content="http://localhost:4000/2018/06/02/stringsasfactors-trouble-in-r.html" />
<meta property="og:site_name" content="Journey of a Bioinformatician" />
<meta property="og:type" content="article" />
<meta property="article:published_time" content="2018-06-02T12:55:00-04:00" />
<script type="application/ld+json">
{"url":"http://localhost:4000/2018/06/02/stringsasfactors-trouble-in-r.html","mainEntityOfPage":{"@type":"WebPage","@id":"http://localhost:4000/2018/06/02/stringsasfactors-trouble-in-r.html"},"author":{"@type":"Person","name":"Aarthi Ramakrishnan"},"description":"Recently, I was reading a counts matrix into R -&gt; counts &lt;- read.table(&quot;htseq_counts_matrix.txt&quot;)&gt; head(counts) Pool5_RNAseq_12_S33_L006 Pool5_RNAseq_19_S34_L006ENSMUSG00000000001 4676 2314ENSMUSG00000000003 0 0ENSMUSG00000000028 558 172ENSMUSG00000000031 25272 6334ENSMUSG00000000037 28 118ENSMUSG00000000049 3 0 Pool5_RNAseq_4_S29_L006 Pool5_RNAseq_5_S30_L006ENSMUSG00000000001 3422 4300ENSMUSG00000000003 0 0ENSMUSG00000000028 328 581ENSMUSG00000000031 7878 22343ENSMUSG00000000037 212 21ENSMUSG00000000049 2 3 Pool5_RNAseq_6_S31_L006 Pool5_RNAseq_7_S32_L006ENSMUSG00000000001 2221 857ENSMUSG00000000003 0 0ENSMUSG00000000028 158 159ENSMUSG00000000031 27124 3655ENSMUSG00000000037 5 32ENSMUSG00000000049 0 0I required the counts for 2 gene IDs only. The 2 IDs were stored in a separate file called &#39;some_genes.txt&#39;. &gt; some_genes &lt;- read.table(&quot;some_genes.txt&quot;)&gt; some_genes &lt;- some_genes$V1&gt; some_genes[1] ENSMUSG00000043587 ENSMUSG00000040276Levels: ENSMUSG00000040276 ENSMUSG00000043587The 2 IDs were ENSMUSG00000043587 and ENSMUSG00000040276. When I tried to subset the counts matrix to retain the counts of just these IDs, something unexpected happened.&gt; some_genes_counts &lt;- counts[some_genes, ]&gt; some_genes_counts Pool5_RNAseq_12_S33_L006 Pool5_RNAseq_19_S34_L006ENSMUSG00000000003 0 0ENSMUSG00000000001 4676 2314 Pool5_RNAseq_4_S29_L006 Pool5_RNAseq_5_S30_L006ENSMUSG00000000003 0 0ENSMUSG00000000001 3422 4300 Pool5_RNAseq_6_S31_L006 Pool5_RNAseq_7_S32_L006ENSMUSG00000000003 0 0ENSMUSG00000000001 2221 857 Instead of obtaining the counts for the 2 IDs, I obtained the counts of 2 totally random genes! When I investigated further -&gt; typeof(some_genes)[1] &quot;integer&quot;The variables in &#39;some_genes&#39; were being converted to integers, and the integers were being used as the row index to subset the counts matrix. This is totally undesirable! But once I included &#39;stringsAsFactors = F&#39; while reading the file into R, the problem was fixed.&nbsp;&gt; some_genes &lt;- read.table(&quot;some_genes.txt&quot;, stringsAsFactors = F)&gt; some_genes &lt;- some_genes$V1&gt; some_genes[1] &quot;ENSMUSG00000043587&quot; &quot;ENSMUSG00000040276&quot;&gt; some_genes_counts &lt;- counts[some_genes, ]&gt; some_genes_counts Pool5_RNAseq_12_S33_L006 Pool5_RNAseq_19_S34_L006ENSMUSG00000043587 385 510ENSMUSG00000040276 5 942 Pool5_RNAseq_4_S29_L006 Pool5_RNAseq_5_S30_L006ENSMUSG00000043587 836 365ENSMUSG00000040276 1021 4 Pool5_RNAseq_6_S31_L006 Pool5_RNAseq_7_S32_L006ENSMUSG00000043587 44 229ENSMUSG00000040276 12 515&gt; typeof(some_genes)[1] &quot;character&quot;This time, I obtained the counts of the right set of IDs. R by default sets stringsAsFactors = T because this setting is convenient while running models using the glm() or lm() function.&nbsp;An alternative to including stringsAsFactors = F for every file you read into R is to just set -&gt; options(stringsAsFactors = FALSE)at the beginning of the R scriptThe following article was also very helpful in getting a better understanding of this problem:https://simplystatistics.org/2015/07/24/stringsasfactors-an-unauthorized-biography/","@type":"BlogPosting","headline":"stringsAsFactors trouble in R","dateModified":"2018-06-02T12:55:00-04:00","datePublished":"2018-06-02T12:55:00-04:00","@context":"http://schema.org"}</script>
<!-- End Jekyll SEO tag -->
<link rel="stylesheet" href="/assets/main.css"><link type="application/atom+xml" rel="alternate" href="http://localhost:4000/feed.xml" title="Journey of a Bioinformatician" /></head>
<body><header class="site-header" role="banner">

  <div class="wrapper"><a class="site-title" rel="author" href="/">Journey of a Bioinformatician</a><nav class="site-nav">
        <input type="checkbox" id="nav-trigger" class="nav-trigger" />
        <label for="nav-trigger">
          <span class="menu-icon">
            <svg viewBox="0 0 18 15" width="18px" height="15px">
              <path d="M18,1.484c0,0.82-0.665,1.484-1.484,1.484H1.484C0.665,2.969,0,2.304,0,1.484l0,0C0,0.665,0.665,0,1.484,0 h15.032C17.335,0,18,0.665,18,1.484L18,1.484z M18,7.516C18,8.335,17.335,9,16.516,9H1.484C0.665,9,0,8.335,0,7.516l0,0 c0-0.82,0.665-1.484,1.484-1.484h15.032C17.335,6.031,18,6.696,18,7.516L18,7.516z M18,13.516C18,14.335,17.335,15,16.516,15H1.484 C0.665,15,0,14.335,0,13.516l0,0c0-0.82,0.665-1.483,1.484-1.483h15.032C17.335,12.031,18,12.695,18,13.516L18,13.516z"/>
            </svg>
          </span>
        </label>

        <div class="trigger"><a class="page-link" href="/about/">About</a></div>
      </nav></div>
</header>
<main class="page-content" aria-label="Content">
      <div class="wrapper">
        <article class="post h-entry" itemscope itemtype="http://schema.org/BlogPosting">

  <header class="post-header">
    <h1 class="post-title p-name" itemprop="name headline">stringsAsFactors trouble in R</h1>
    <p class="post-meta">
      <time class="dt-published" datetime="2018-06-02T12:55:00-04:00" itemprop="datePublished">Jun 2, 2018
      </time>â€¢ <span itemprop="author" itemscope itemtype="http://schema.org/Person"><span class="p-author h-card" itemprop="name">Aarthi Ramakrishnan</span></span></p>
  </header>

  <div class="post-content e-content" itemprop="articleBody">
    <div dir="ltr" style="text-align: left;" trbidi="on"><span style="font-family: inherit;">Recently, I was reading a counts matrix into R -</span><br /><!-- HTML generated using hilite.me --><br /><div style="background: #ffffff; border-width: 0.1em 0.1em 0.1em 0.8em; border: solid gray; overflow: auto; padding: 0.2em 0.6em; width: auto;"><pre style="line-height: 125%; margin: 0;"><span style="color: #c65d09; font-weight: bold;">&gt; </span>counts <span style="color: #333333;">&lt;-</span> read.table(<span style="background-color: #fff0f0;">"htseq_counts_matrix.txt"</span>)<br /><span style="color: #c65d09; font-weight: bold;">&gt; </span>head(counts)<br /><span style="color: #888888;">                   Pool5_RNAseq_12_S33_L006 Pool5_RNAseq_19_S34_L006</span><br /><span style="color: #888888;">ENSMUSG00000000001                     4676                     2314</span><br /><span style="color: #888888;">ENSMUSG00000000003                        0                        0</span><br /><span style="color: #888888;">ENSMUSG00000000028                      558                      172</span><br /><span style="color: #888888;">ENSMUSG00000000031                    25272                     6334</span><br /><span style="color: #888888;">ENSMUSG00000000037                       28                      118</span><br /><span style="color: #888888;">ENSMUSG00000000049                        3                        0</span><br /><span style="color: #888888;">                   Pool5_RNAseq_4_S29_L006 Pool5_RNAseq_5_S30_L006</span><br /><span style="color: #888888;">ENSMUSG00000000001                    3422                    4300</span><br /><span style="color: #888888;">ENSMUSG00000000003                       0                       0</span><br /><span style="color: #888888;">ENSMUSG00000000028                     328                     581</span><br /><span style="color: #888888;">ENSMUSG00000000031                    7878                   22343</span><br /><span style="color: #888888;">ENSMUSG00000000037                     212                      21</span><br /><span style="color: #888888;">ENSMUSG00000000049                       2                       3</span><br /><span style="color: #888888;">                   Pool5_RNAseq_6_S31_L006 Pool5_RNAseq_7_S32_L006</span><br /><span style="color: #888888;">ENSMUSG00000000001                    2221                     857</span><br /><span style="color: #888888;">ENSMUSG00000000003                       0                       0</span><br /><span style="color: #888888;">ENSMUSG00000000028                     158                     159</span><br /><span style="color: #888888;">ENSMUSG00000000031                   27124                    3655</span><br /><span style="color: #888888;">ENSMUSG00000000037                       5                      32</span><br /><span style="color: #888888;">ENSMUSG00000000049                       0                       0</span><br /></pre></div><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">I required the counts for 2 gene IDs only. The 2 IDs were stored in a separate file called 'some_genes.txt'.</span><br /><span style="font-family: inherit;"><br /></span><!-- HTML generated using hilite.me --> <br /><div style="background: #ffffff; border-width: 0.1em 0.1em 0.1em 0.8em; border: solid gray; overflow: auto; padding: 0.2em 0.6em; width: auto;"><pre style="line-height: 125%; margin: 0;"><span style="color: #c65d09; font-weight: bold;">&gt; </span>some_genes <span style="color: #333333;">&lt;-</span> read.table(<span style="background-color: #fff0f0;">"some_genes.txt"</span>)<br /><span style="color: #c65d09; font-weight: bold;">&gt; </span>some_genes <span style="color: #333333;">&lt;-</span> some_genes<span style="color: #333333;">$</span>V1<br /><span style="color: #c65d09; font-weight: bold;">&gt; </span>some_genes<br /><span style="color: #888888;">[1] ENSMUSG00000043587 ENSMUSG00000040276</span><br /><span style="color: #888888;">Levels: ENSMUSG00000040276 ENSMUSG00000043587</span><br /></pre></div><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">The 2 IDs were ENSMUSG00000043587 and ENSMUSG00000040276. When I tried to subset the counts matrix to retain the counts of just these IDs, something unexpected happened.</span><br /><span style="font-family: inherit;"><br /></span><!-- HTML generated using hilite.me --><br /><div style="background: #ffffff; border-width: 0.1em 0.1em 0.1em 0.8em; border: solid gray; overflow: auto; padding: 0.2em 0.6em; width: auto;"><pre style="line-height: 125%; margin: 0;"><span style="color: #c65d09; font-weight: bold;">&gt; </span>some_genes_counts <span style="color: #333333;">&lt;-</span> counts[some_genes, ]<br /><span style="color: #c65d09; font-weight: bold;">&gt; </span>some_genes_counts<br /><span style="color: #888888;">                   Pool5_RNAseq_12_S33_L006 Pool5_RNAseq_19_S34_L006</span><br /><span style="color: #888888;">ENSMUSG00000000003                        0                        0</span><br /><span style="color: #888888;">ENSMUSG00000000001                     4676                     2314</span><br /><span style="color: #888888;">                   Pool5_RNAseq_4_S29_L006 Pool5_RNAseq_5_S30_L006</span><br /><span style="color: #888888;">ENSMUSG00000000003                       0                       0</span><br /><span style="color: #888888;">ENSMUSG00000000001                    3422                    4300</span><br /><span style="color: #888888;">                   Pool5_RNAseq_6_S31_L006 Pool5_RNAseq_7_S32_L006</span><br /><span style="color: #888888;">ENSMUSG00000000003                       0                       0</span><br /><span style="color: #888888;">ENSMUSG00000000001                    2221                     857</span><br /></pre></div><span style="font-family: inherit;"><br /></span> <span style="font-family: inherit;">Instead of obtaining the counts for the 2 IDs, I obtained the counts of 2 totally random genes! When I investigated further -</span><br /><span style="font-family: inherit;"><br /></span><!-- HTML generated using hilite.me --><br /><div style="background: #ffffff; border-width: 0.1em 0.1em 0.1em 0.8em; border: solid gray; overflow: auto; padding: 0.2em 0.6em; width: auto;"><pre style="line-height: 125%; margin: 0;"><span style="color: #c65d09; font-weight: bold;">&gt; </span>typeof(some_genes)<br /><span style="color: #888888;">[1] "integer"</span><br /></pre></div><br /><span style="font-family: inherit;">The variables in 'some_genes' were being converted to integers, and the integers were being used as the row index to subset the counts matrix. This is totally undesirable! But once I included 'stringsAsFactors = F' while reading the file into R, the problem was fixed.&nbsp;</span><br /><span style="font-family: inherit;"><br /></span><!-- HTML generated using hilite.me --><br /><div style="background: #ffffff; border-width: 0.1em 0.1em 0.1em 0.8em; border: solid gray; overflow: auto; padding: 0.2em 0.6em; width: auto;"><pre style="line-height: 125%; margin: 0;"><span style="color: #c65d09; font-weight: bold;">&gt; </span>some_genes <span style="color: #333333;">&lt;-</span> read.table(<span style="background-color: #fff0f0;">"some_genes.txt"</span>, stringsAsFactors <span style="color: #333333;">=</span> <span style="color: #008800; font-weight: bold;">F</span>)<br /><span style="color: #c65d09; font-weight: bold;">&gt; </span>some_genes <span style="color: #333333;">&lt;-</span> some_genes<span style="color: #333333;">$</span>V1<br /><span style="color: #c65d09; font-weight: bold;">&gt; </span>some_genes<br /><span style="color: #888888;">[1] "ENSMUSG00000043587" "ENSMUSG00000040276"</span><br /><span style="color: #c65d09; font-weight: bold;">&gt; </span>some_genes_counts <span style="color: #333333;">&lt;-</span> counts[some_genes, ]<br /><span style="color: #c65d09; font-weight: bold;">&gt; </span>some_genes_counts<br /><span style="color: #888888;">                   Pool5_RNAseq_12_S33_L006 Pool5_RNAseq_19_S34_L006</span><br /><span style="color: #888888;">ENSMUSG00000043587                      385                      510</span><br /><span style="color: #888888;">ENSMUSG00000040276                        5                      942</span><br /><span style="color: #888888;">                   Pool5_RNAseq_4_S29_L006 Pool5_RNAseq_5_S30_L006</span><br /><span style="color: #888888;">ENSMUSG00000043587                     836                     365</span><br /><span style="color: #888888;">ENSMUSG00000040276                    1021                       4</span><br /><span style="color: #888888;">                   Pool5_RNAseq_6_S31_L006 Pool5_RNAseq_7_S32_L006</span><br /><span style="color: #888888;">ENSMUSG00000043587                      44                     229</span><br /><span style="color: #888888;">ENSMUSG00000040276                      12                     515</span><br /><span style="color: #c65d09; font-weight: bold;">&gt; </span>typeof(some_genes)<br /><span style="color: #888888;">[1] "character"</span><br /></pre></div><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">This time, I obtained the counts of the right set of IDs. R by default sets stringsAsFactors = T because this setting is convenient while running models using the glm() or lm() function.&nbsp;</span><br /><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">An alternative to including stringsAsFactors = F for every file you read into R is to just set -</span><br /><span style="font-family: inherit;"><br /></span><!-- HTML generated using hilite.me --><br /><div style="background: #ffffff; border-width: 0.1em 0.1em 0.1em 0.8em; border: solid gray; overflow: auto; padding: 0.2em 0.6em; width: auto;"><pre style="line-height: 125%; margin: 0;"><span style="color: #c65d09; font-weight: bold;">&gt; </span>options(stringsAsFactors <span style="color: #333333;">=</span> <span style="color: #008800; font-weight: bold;">FALSE</span>)<br /></pre></div><br />at the beginning of the R script<br /><br />The following article was also very helpful in getting a better understanding of this problem:<br /><br /><a href="https://simplystatistics.org/2015/07/24/stringsasfactors-an-unauthorized-biography/">https://simplystatistics.org/2015/07/24/stringsasfactors-an-unauthorized-biography/</a></div>
  </div><a class="u-url" href="/2018/06/02/stringsasfactors-trouble-in-r.html" hidden></a>
</article>

      </div>
    </main><footer class="site-footer h-card">
  <data class="u-url" href="/"></data>

  <div class="wrapper">

    <h2 class="footer-heading">Journey of a Bioinformatician</h2>

    <div class="footer-col-wrapper">
      <div class="footer-col footer-col-1">
        <ul class="contact-list">
          <li class="p-name">Journey of a Bioinformatician</li><li><a class="u-email" href="mailto:aarthir239@gmail.com">aarthir239@gmail.com</a></li></ul>
      </div>

      <div class="footer-col footer-col-2"><ul class="social-media-list"><li><a href="https://github.com/aartrama"><svg class="svg-icon"><use xlink:href="/assets/minima-social-icons.svg#github"></use></svg> <span class="username">aartrama</span></a></li></ul>
</div>

      <div class="footer-col footer-col-3">
        <p>Aarthi Ramakrishnan&#39;s blog</p>
      </div>
    </div>

  </div>

</footer>
</body>

</html>
