<!DOCTYPE html>
<html lang="en"><head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1"><!-- Begin Jekyll SEO tag v2.5.0 -->
<title>Visualization of ChIP-seq data | Journey of a Bioinformatician</title>
<meta name="generator" content="Jekyll v3.8.3" />
<meta property="og:title" content="Visualization of ChIP-seq data" />
<meta name="author" content="Aarthi Ramakrishnan" />
<meta property="og:locale" content="en_US" />
<meta name="description" content="For visualization of ChIP-seq datasets, it is always recommended to convert bedgraph files obtained from MACS to bigwig format files prior to visualization on the UCSC genome browser. Following are the steps to be followed for the conversion:&nbsp;1. To sort bedgraph output files from MACS, use the following commandLC_COLLATE=C sort -k1,1 -k2,2n filename.bdg &gt; sorted.filename.bdg 2. (OPTIONAL) Once all files (control and treated) have been sorted, the next step is to remove the last line from each of the sorted files. This can be done using the following code:head -n -1 sorted.filename.bdg &gt; temp.sorted.filename.bdg ; mv temp.sorted.filename.bdg sorted.filename.bdg *credits - http://stackoverflow.com/questions/4881930/remove-the-last-line-from-a-file-in-bash3. (OPTIONAL) Next, the sorted files should be clipped using the tool bedClip. Following is the command to be used:bedClip sorted.filename.bdg hg38.chrom.sizes sorted.clipped.filename.bdgWhere hg38.chrom.sizes is a tab delimited text file that contains chromosome name (without &#39;chr&#39; in the beginning of each line) in the first column and chromosome size in the second column. This information can be obtained from SAM files of the same datasets generated using a reference from ENSEMBL .&nbsp;4. (OPTIONAL) The next step would be to add &#39;chr&#39; to each line in all bedgraph files, as UCSC file format requires &#39;chr&#39; to be present before each chromosome name (for chromosomes 1-22, X and Y). I do this using a custom python script as follows:&quot;&quot;&quot;This program is for addingchr to each line of the bdgfile if line starts with1-22, x or y&quot;&quot;&quot;# INSERTION OF FILE NAMESfiles = [&#39;sorted.clipped.filename.bdg&#39;, \ &#39;sorted.clipped.filename1.bdg&#39;, \ &#39;sorted.clipped.filename2.bdg&#39;, \ &#39;sorted.clipped.filename3.bdg&#39;]# CREATION OF LIST OF CHROMOSOMESlist1 = [&#39;X&#39;, &#39;Y&#39;, &#39;MT&#39;]for i in range(1, 23): list1.append(str(i))for file in files: with open(file, &quot;r&quot;) as f, \ open(&quot;chr.&quot;+file, &quot;w&quot;) as outfile: for lines in f: lines = lines.strip().split() if lines[0] in list1: print&gt;&gt;outfile, &quot;chr&quot;+&quot;\t&quot;.join(lines)5. The final step is the conversion of bedgraph file to bigwig file. To do this, bedGraphToBigWig tool from UCSC can be utilized. Following is the command (note that hg38.chrom.sizes should have &#39;chr&#39; at the beginning of each chromosome here):bedGraphToBigWig sorted.clipped.filename.bdg hg38.chrom.sizes filename.bwYou can now upload the bw files in a web server and provide the link to these files in &#39;custom tracks&#39; of UCSC Genome Browser." />
<meta property="og:description" content="For visualization of ChIP-seq datasets, it is always recommended to convert bedgraph files obtained from MACS to bigwig format files prior to visualization on the UCSC genome browser. Following are the steps to be followed for the conversion:&nbsp;1. To sort bedgraph output files from MACS, use the following commandLC_COLLATE=C sort -k1,1 -k2,2n filename.bdg &gt; sorted.filename.bdg 2. (OPTIONAL) Once all files (control and treated) have been sorted, the next step is to remove the last line from each of the sorted files. This can be done using the following code:head -n -1 sorted.filename.bdg &gt; temp.sorted.filename.bdg ; mv temp.sorted.filename.bdg sorted.filename.bdg *credits - http://stackoverflow.com/questions/4881930/remove-the-last-line-from-a-file-in-bash3. (OPTIONAL) Next, the sorted files should be clipped using the tool bedClip. Following is the command to be used:bedClip sorted.filename.bdg hg38.chrom.sizes sorted.clipped.filename.bdgWhere hg38.chrom.sizes is a tab delimited text file that contains chromosome name (without &#39;chr&#39; in the beginning of each line) in the first column and chromosome size in the second column. This information can be obtained from SAM files of the same datasets generated using a reference from ENSEMBL .&nbsp;4. (OPTIONAL) The next step would be to add &#39;chr&#39; to each line in all bedgraph files, as UCSC file format requires &#39;chr&#39; to be present before each chromosome name (for chromosomes 1-22, X and Y). I do this using a custom python script as follows:&quot;&quot;&quot;This program is for addingchr to each line of the bdgfile if line starts with1-22, x or y&quot;&quot;&quot;# INSERTION OF FILE NAMESfiles = [&#39;sorted.clipped.filename.bdg&#39;, \ &#39;sorted.clipped.filename1.bdg&#39;, \ &#39;sorted.clipped.filename2.bdg&#39;, \ &#39;sorted.clipped.filename3.bdg&#39;]# CREATION OF LIST OF CHROMOSOMESlist1 = [&#39;X&#39;, &#39;Y&#39;, &#39;MT&#39;]for i in range(1, 23): list1.append(str(i))for file in files: with open(file, &quot;r&quot;) as f, \ open(&quot;chr.&quot;+file, &quot;w&quot;) as outfile: for lines in f: lines = lines.strip().split() if lines[0] in list1: print&gt;&gt;outfile, &quot;chr&quot;+&quot;\t&quot;.join(lines)5. The final step is the conversion of bedgraph file to bigwig file. To do this, bedGraphToBigWig tool from UCSC can be utilized. Following is the command (note that hg38.chrom.sizes should have &#39;chr&#39; at the beginning of each chromosome here):bedGraphToBigWig sorted.clipped.filename.bdg hg38.chrom.sizes filename.bwYou can now upload the bw files in a web server and provide the link to these files in &#39;custom tracks&#39; of UCSC Genome Browser." />
<link rel="canonical" href="http://localhost:4000/2017/05/07/visualization-of-chip-seq-data.html" />
<meta property="og:url" content="http://localhost:4000/2017/05/07/visualization-of-chip-seq-data.html" />
<meta property="og:site_name" content="Journey of a Bioinformatician" />
<meta property="og:type" content="article" />
<meta property="article:published_time" content="2017-05-07T19:42:00-04:00" />
<script type="application/ld+json">
{"url":"http://localhost:4000/2017/05/07/visualization-of-chip-seq-data.html","mainEntityOfPage":{"@type":"WebPage","@id":"http://localhost:4000/2017/05/07/visualization-of-chip-seq-data.html"},"author":{"@type":"Person","name":"Aarthi Ramakrishnan"},"description":"For visualization of ChIP-seq datasets, it is always recommended to convert bedgraph files obtained from MACS to bigwig format files prior to visualization on the UCSC genome browser. Following are the steps to be followed for the conversion:&nbsp;1. To sort bedgraph output files from MACS, use the following commandLC_COLLATE=C sort -k1,1 -k2,2n filename.bdg &gt; sorted.filename.bdg 2. (OPTIONAL) Once all files (control and treated) have been sorted, the next step is to remove the last line from each of the sorted files. This can be done using the following code:head -n -1 sorted.filename.bdg &gt; temp.sorted.filename.bdg ; mv temp.sorted.filename.bdg sorted.filename.bdg *credits - http://stackoverflow.com/questions/4881930/remove-the-last-line-from-a-file-in-bash3. (OPTIONAL) Next, the sorted files should be clipped using the tool bedClip. Following is the command to be used:bedClip sorted.filename.bdg hg38.chrom.sizes sorted.clipped.filename.bdgWhere hg38.chrom.sizes is a tab delimited text file that contains chromosome name (without &#39;chr&#39; in the beginning of each line) in the first column and chromosome size in the second column. This information can be obtained from SAM files of the same datasets generated using a reference from ENSEMBL .&nbsp;4. (OPTIONAL) The next step would be to add &#39;chr&#39; to each line in all bedgraph files, as UCSC file format requires &#39;chr&#39; to be present before each chromosome name (for chromosomes 1-22, X and Y). I do this using a custom python script as follows:&quot;&quot;&quot;This program is for addingchr to each line of the bdgfile if line starts with1-22, x or y&quot;&quot;&quot;# INSERTION OF FILE NAMESfiles = [&#39;sorted.clipped.filename.bdg&#39;, \\ &#39;sorted.clipped.filename1.bdg&#39;, \\ &#39;sorted.clipped.filename2.bdg&#39;, \\ &#39;sorted.clipped.filename3.bdg&#39;]# CREATION OF LIST OF CHROMOSOMESlist1 = [&#39;X&#39;, &#39;Y&#39;, &#39;MT&#39;]for i in range(1, 23): list1.append(str(i))for file in files: with open(file, &quot;r&quot;) as f, \\ open(&quot;chr.&quot;+file, &quot;w&quot;) as outfile: for lines in f: lines = lines.strip().split() if lines[0] in list1: print&gt;&gt;outfile, &quot;chr&quot;+&quot;\\t&quot;.join(lines)5. The final step is the conversion of bedgraph file to bigwig file. To do this, bedGraphToBigWig tool from UCSC can be utilized. Following is the command (note that hg38.chrom.sizes should have &#39;chr&#39; at the beginning of each chromosome here):bedGraphToBigWig sorted.clipped.filename.bdg hg38.chrom.sizes filename.bwYou can now upload the bw files in a web server and provide the link to these files in &#39;custom tracks&#39; of UCSC Genome Browser.","@type":"BlogPosting","headline":"Visualization of ChIP-seq data","dateModified":"2017-05-07T19:42:00-04:00","datePublished":"2017-05-07T19:42:00-04:00","@context":"http://schema.org"}</script>
<!-- End Jekyll SEO tag -->
<link rel="stylesheet" href="/assets/main.css"><link type="application/atom+xml" rel="alternate" href="http://localhost:4000/feed.xml" title="Journey of a Bioinformatician" /></head>
<body><header class="site-header" role="banner">

  <div class="wrapper"><a class="site-title" rel="author" href="/">Journey of a Bioinformatician</a><nav class="site-nav">
        <input type="checkbox" id="nav-trigger" class="nav-trigger" />
        <label for="nav-trigger">
          <span class="menu-icon">
            <svg viewBox="0 0 18 15" width="18px" height="15px">
              <path d="M18,1.484c0,0.82-0.665,1.484-1.484,1.484H1.484C0.665,2.969,0,2.304,0,1.484l0,0C0,0.665,0.665,0,1.484,0 h15.032C17.335,0,18,0.665,18,1.484L18,1.484z M18,7.516C18,8.335,17.335,9,16.516,9H1.484C0.665,9,0,8.335,0,7.516l0,0 c0-0.82,0.665-1.484,1.484-1.484h15.032C17.335,6.031,18,6.696,18,7.516L18,7.516z M18,13.516C18,14.335,17.335,15,16.516,15H1.484 C0.665,15,0,14.335,0,13.516l0,0c0-0.82,0.665-1.483,1.484-1.483h15.032C17.335,12.031,18,12.695,18,13.516L18,13.516z"/>
            </svg>
          </span>
        </label>

        <div class="trigger"><a class="page-link" href="/about/">About</a></div>
      </nav></div>
</header>
<main class="page-content" aria-label="Content">
      <div class="wrapper">
        <article class="post h-entry" itemscope itemtype="http://schema.org/BlogPosting">

  <header class="post-header">
    <h1 class="post-title p-name" itemprop="name headline">Visualization of ChIP-seq data</h1>
    <p class="post-meta">
      <time class="dt-published" datetime="2017-05-07T19:42:00-04:00" itemprop="datePublished">May 7, 2017
      </time>â€¢ <span itemprop="author" itemscope itemtype="http://schema.org/Person"><span class="p-author h-card" itemprop="name">Aarthi Ramakrishnan</span></span></p>
  </header>

  <div class="post-content e-content" itemprop="articleBody">
    <div dir="ltr" style="text-align: left;" trbidi="on"><span style="font-family: inherit;">For visualization of ChIP-seq datasets, it is always recommended to convert bedgraph files obtained from MACS to bigwig format files prior to visualization on the UCSC genome browser. Following are the steps to be followed for the conversion:&nbsp;</span><br /><span style="font-family: inherit;"><span style="font-family: inherit;"><br /></span><span style="font-family: inherit;">1. To sort bedgraph output files from MACS, use the following command</span></span><br /><span style="font-family: inherit;"><span style="font-family: inherit;"><br /></span><!-- HTML generated using hilite.me --></span><br /><div style="background: #ffffff; border-width: 0.1em 0.1em 0.1em 0.8em; border: solid gray; overflow: auto; padding: 0.2em 0.6em; width: auto;"><pre style="line-height: 125%; margin: 0;"><span style="font-family: inherit;"><span style="color: #996633;">LC_COLLATE</span><span style="color: #333333;">=</span>C sort -k1,1 -k2,2n filename.bdg &gt; sorted.filename.bdg<br /></span></pre></div><span style="font-family: inherit;"><span style="font-family: inherit;"><br /></span> <span style="font-family: inherit;">2. (OPTIONAL) Once all files (control and treated) have been sorted, the next step is to remove the last line from each of the sorted files. This can be done using the following code:</span></span><br /><span style="font-family: inherit;"><br /></span><!-- HTML generated using hilite.me --><br /><div style="background: #ffffff; border-width: 0.1em 0.1em 0.1em 0.8em; border: solid gray; overflow: auto; padding: 0.2em 0.6em; width: auto;"><pre style="line-height: 125%; margin: 0;"><span style="font-family: inherit;">head -n -1 sorted.filename.bdg &gt; temp.sorted.filename.bdg ; mv temp.sorted.filename.bdg sorted.filename.bdg<br /></span></pre></div><span style="font-family: inherit;"><br /></span> <span style="color: #cccccc; font-family: inherit;">*credits - http://stackoverflow.com/questions/4881930/remove-the-last-line-from-a-file-in-bash</span><br /><span style="font-family: inherit;"><span style="font-family: inherit;"><br /></span></span><br /><span style="font-family: inherit;">3. (OPTIONAL) Next, the sorted files should be clipped using the tool bedClip. Following is the command to be used:</span><br /><span style="font-family: inherit;"><br /></span><!-- HTML generated using hilite.me --><br /><div style="background: #ffffff; border-width: 0.1em 0.1em 0.1em 0.8em; border: solid gray; overflow: auto; padding: 0.2em 0.6em; width: auto;"><pre style="line-height: 125%; margin: 0;">bedClip sorted.filename.bdg hg38.chrom.sizes sorted.clipped.filename.bdg<br /></pre></div><br /><span style="font-family: inherit;"><span style="font-family: inherit;">Where hg38.chrom.sizes is a tab delimited text file that contains chromosome name (without 'chr' in the beginning of each line) in the first column and chromosome size in the second column. This information can be obtained from SAM files of the same datasets generated using a reference from ENSEMBL .&nbsp;</span></span><br /><span style="font-family: inherit;"><span style="font-family: inherit;"><br /></span></span><br /><span style="font-family: inherit;">4. (OPTIONAL) The next step would be to add 'chr' to each line in all bedgraph files, as UCSC file format requires 'chr' to be present before each chromosome name (for chromosomes 1-22, X and Y). I do this using a custom python script as follows:</span><br /><span style="font-family: inherit;"><br /></span><!-- HTML generated using hilite.me --><br /><div style="background: #ffffff; border-width: 0.1em 0.1em 0.1em 0.8em; border: solid gray; overflow: auto; padding: 0.2em 0.6em; width: auto;"><pre style="line-height: 125%; margin: 0;"><span style="color: #dd4422;">"""</span><br /><span style="color: #dd4422;">This program is for adding</span><br /><span style="color: #dd4422;">chr to each line of the bdg</span><br /><span style="color: #dd4422;">file if line starts with</span><br /><span style="color: #dd4422;">1-22, x or y</span><br /><span style="color: #dd4422;">"""</span><br /><br /><br /><span style="color: #888888;"># INSERTION OF FILE NAMES</span><br /><br />files <span style="color: #333333;">=</span> [<span style="background-color: #fff0f0;">'sorted.clipped.filename.bdg'</span>, \<br /> <span style="background-color: #fff0f0;">'sorted.clipped.filename1.bdg'</span>, \<br /> <span style="background-color: #fff0f0;">'sorted.clipped.filename2.bdg'</span>, \<br /> <span style="background-color: #fff0f0;">'sorted.clipped.filename3.bdg'</span>]<br /><br /><span style="color: #888888;"># CREATION OF LIST OF CHROMOSOMES</span><br /><br />list1 <span style="color: #333333;">=</span> [<span style="background-color: #fff0f0;">'X'</span>, <span style="background-color: #fff0f0;">'Y', 'MT'</span>]<br /><br /><span style="color: #008800; font-weight: bold;">for</span> i <span style="color: black; font-weight: bold;">in</span> <span style="color: #007020;">range</span>(<span style="color: #0000dd; font-weight: bold;">1</span>, <span style="color: #0000dd; font-weight: bold;">23</span>):<br />    list1<span style="color: #333333;">.</span>append(<span style="color: #007020;">str</span>(i))<br /><br /><span style="color: #008800; font-weight: bold;">for</span> <span style="color: #007020;">file</span> <span style="color: black; font-weight: bold;">in</span> files:<br />    <span style="color: #008800; font-weight: bold;">with</span> <span style="color: #007020;">open</span>(<span style="color: #007020;">file</span>, <span style="background-color: #fff0f0;">"r"</span>) <span style="color: #008800; font-weight: bold;">as</span> f, \<br />    <span style="color: #007020;">open</span>(<span style="background-color: #fff0f0;">"chr."</span><span style="color: #333333;">+</span><span style="color: #007020;">file</span>, <span style="background-color: #fff0f0;">"w"</span>) <span style="color: #008800; font-weight: bold;">as</span> outfile:<br />    <span style="color: #008800; font-weight: bold;">for</span> lines <span style="color: black; font-weight: bold;">in</span> f:<br />        lines <span style="color: #333333;">=</span> lines<span style="color: #333333;">.</span>strip()<span style="color: #333333;">.</span>split()<br />        <span style="color: #008800; font-weight: bold;">if</span> lines[<span style="color: #0000dd; font-weight: bold;">0</span>] <span style="color: black; font-weight: bold;">in</span> list1:<br />            <span style="color: #008800; font-weight: bold;">print</span><span style="color: #333333;">&gt;&gt;</span>outfile, <span style="background-color: #fff0f0;">"chr"</span><span style="color: #333333;">+</span><span style="background-color: #fff0f0;">"</span><span style="background-color: #fff0f0; color: #666666; font-weight: bold;">\t</span><span style="background-color: #fff0f0;">"</span><span style="color: #333333;">.</span>join(lines)<br /></pre></div><br /><span style="font-family: inherit;"><span style="font-family: inherit;">5. The final step is the conversion of bedgraph file to bigwig file. To do this, bedGraphToBigWig tool from UCSC can be utilized. Following is the command (note that hg38.chrom.sizes should have 'chr' at the beginning of each chromosome here):</span></span><br /><span style="font-family: inherit;"><span style="font-family: inherit;"><br /></span></span><!-- HTML generated using hilite.me --><br /><div style="background: #ffffff; border-width: 0.1em 0.1em 0.1em 0.8em; border: solid gray; overflow: auto; padding: 0.2em 0.6em; width: auto;"><pre style="line-height: 125%; margin: 0;">bedGraphToBigWig sorted.clipped.filename.bdg hg38.chrom.sizes filename.bw<br /></pre></div><br /><span style="font-family: inherit;">You can now upload the bw files in a web server and provide the link to these files in 'custom tracks' of UCSC Genome Browser.</span></div>
  </div><a class="u-url" href="/2017/05/07/visualization-of-chip-seq-data.html" hidden></a>
</article>

      </div>
    </main><footer class="site-footer h-card">
  <data class="u-url" href="/"></data>

  <div class="wrapper">

    <h2 class="footer-heading">Journey of a Bioinformatician</h2>

    <div class="footer-col-wrapper">
      <div class="footer-col footer-col-1">
        <ul class="contact-list">
          <li class="p-name">Journey of a Bioinformatician</li><li><a class="u-email" href="mailto:aarthir239@gmail.com">aarthir239@gmail.com</a></li></ul>
      </div>

      <div class="footer-col footer-col-2"><ul class="social-media-list"><li><a href="https://github.com/aartrama"><svg class="svg-icon"><use xlink:href="/assets/minima-social-icons.svg#github"></use></svg> <span class="username">aartrama</span></a></li></ul>
</div>

      <div class="footer-col footer-col-3">
        <p>Aarthi Ramakrishnan&#39;s blog</p>
      </div>
    </div>

  </div>

</footer>
</body>

</html>
