<!DOCTYPE html>
<html lang="en"><head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1"><!-- Begin Jekyll SEO tag v2.5.0 -->
<title>NGS Data analysis tips | Journey of a Bioinformatician</title>
<meta name="generator" content="Jekyll v3.8.3" />
<meta property="og:title" content="NGS Data analysis tips" />
<meta name="author" content="Aarthi Ramakrishnan" />
<meta property="og:locale" content="en_US" />
<meta name="description" content="Following are some suggestions while analyzing NGS datasets -1) NEVER delete the outputs of a particular step if the output generation has failed for SOME of the samples. Complete running the pipeline only for the samples which have failed. It might be tempting to delete everything and start all over again (for that particular step), but NEVER do that! It is a waste of your time and resource. Value what you already have! You never know, the next new set of runs might get stuck on queue on the cluster environment. You can always process the remaining samples later on, as managing a small number of samples is easier than managing a large number of samples.2) Before submitting multiple jobs on a cluster, submit a job for one sample first. Also, submit by setting a small walltime. If that works, submit the remaining jobs.3) On a cluster environment, leaving a job running is better than killing a job. Killing a job penalizes your account, and your subsequent jobs are accepted with delays.4) Whenever you cannot get a bioinformatics tool working, try the following:- Read the manual/command line options and check if you have missed something - Check for spelling mistakes when you manually edit files. AVOID MANUALLY EDITING FILES as much as possible in the first place!- Check the file format of your input files. For example, format your BED files to have 6 fields instead of 3- Check if &#39;chr1&#39; is present in one input file and &#39;1&#39; is present in the other - Email the authors/google the query- Look at the tool&#39;s source code 5) To check whether a bam file is sorted, if you are able to index the file using samtools index {file name}, then the file is sorted. The headers of the sam file are unreliable. The older versions of samtools report SO:unsorted even though the file could be actually sorted!6) Fold change and log fold change are 2 different terms. Do not mistake one for the other.7) Merging fastq files is not a good idea. Instead, merging the bam files is better.8) Whenever you would like to visualize expression data as a heatmap, it is best to log transform the data before visualization. Vst or rlog transform can be used.9) Automate your process wherever possible! Accomplishing tasks manually is not only error prone, but also a waste of time.10) DO NOT copy a piece of code from the terminal window, paste it somewhere else and try to run it. The format of the copied code would have changed." />
<meta property="og:description" content="Following are some suggestions while analyzing NGS datasets -1) NEVER delete the outputs of a particular step if the output generation has failed for SOME of the samples. Complete running the pipeline only for the samples which have failed. It might be tempting to delete everything and start all over again (for that particular step), but NEVER do that! It is a waste of your time and resource. Value what you already have! You never know, the next new set of runs might get stuck on queue on the cluster environment. You can always process the remaining samples later on, as managing a small number of samples is easier than managing a large number of samples.2) Before submitting multiple jobs on a cluster, submit a job for one sample first. Also, submit by setting a small walltime. If that works, submit the remaining jobs.3) On a cluster environment, leaving a job running is better than killing a job. Killing a job penalizes your account, and your subsequent jobs are accepted with delays.4) Whenever you cannot get a bioinformatics tool working, try the following:- Read the manual/command line options and check if you have missed something - Check for spelling mistakes when you manually edit files. AVOID MANUALLY EDITING FILES as much as possible in the first place!- Check the file format of your input files. For example, format your BED files to have 6 fields instead of 3- Check if &#39;chr1&#39; is present in one input file and &#39;1&#39; is present in the other - Email the authors/google the query- Look at the tool&#39;s source code 5) To check whether a bam file is sorted, if you are able to index the file using samtools index {file name}, then the file is sorted. The headers of the sam file are unreliable. The older versions of samtools report SO:unsorted even though the file could be actually sorted!6) Fold change and log fold change are 2 different terms. Do not mistake one for the other.7) Merging fastq files is not a good idea. Instead, merging the bam files is better.8) Whenever you would like to visualize expression data as a heatmap, it is best to log transform the data before visualization. Vst or rlog transform can be used.9) Automate your process wherever possible! Accomplishing tasks manually is not only error prone, but also a waste of time.10) DO NOT copy a piece of code from the terminal window, paste it somewhere else and try to run it. The format of the copied code would have changed." />
<link rel="canonical" href="http://localhost:4000/2017/12/23/ngs-data-analysis-tips.html" />
<meta property="og:url" content="http://localhost:4000/2017/12/23/ngs-data-analysis-tips.html" />
<meta property="og:site_name" content="Journey of a Bioinformatician" />
<meta property="og:type" content="article" />
<meta property="article:published_time" content="2017-12-23T16:58:00-05:00" />
<script type="application/ld+json">
{"url":"http://localhost:4000/2017/12/23/ngs-data-analysis-tips.html","mainEntityOfPage":{"@type":"WebPage","@id":"http://localhost:4000/2017/12/23/ngs-data-analysis-tips.html"},"author":{"@type":"Person","name":"Aarthi Ramakrishnan"},"description":"Following are some suggestions while analyzing NGS datasets -1) NEVER delete the outputs of a particular step if the output generation has failed for SOME of the samples. Complete running the pipeline only for the samples which have failed. It might be tempting to delete everything and start all over again (for that particular step), but NEVER do that! It is a waste of your time and resource. Value what you already have! You never know, the next new set of runs might get stuck on queue on the cluster environment. You can always process the remaining samples later on, as managing a small number of samples is easier than managing a large number of samples.2) Before submitting multiple jobs on a cluster, submit a job for one sample first. Also, submit by setting a small walltime. If that works, submit the remaining jobs.3) On a cluster environment, leaving a job running is better than killing a job. Killing a job penalizes your account, and your subsequent jobs are accepted with delays.4) Whenever you cannot get a bioinformatics tool working, try the following:- Read the manual/command line options and check if you have missed something - Check for spelling mistakes when you manually edit files. AVOID MANUALLY EDITING FILES as much as possible in the first place!- Check the file format of your input files. For example, format your BED files to have 6 fields instead of 3- Check if &#39;chr1&#39; is present in one input file and &#39;1&#39; is present in the other - Email the authors/google the query- Look at the tool&#39;s source code 5) To check whether a bam file is sorted, if you are able to index the file using samtools index {file name}, then the file is sorted. The headers of the sam file are unreliable. The older versions of samtools report SO:unsorted even though the file could be actually sorted!6) Fold change and log fold change are 2 different terms. Do not mistake one for the other.7) Merging fastq files is not a good idea. Instead, merging the bam files is better.8) Whenever you would like to visualize expression data as a heatmap, it is best to log transform the data before visualization. Vst or rlog transform can be used.9) Automate your process wherever possible! Accomplishing tasks manually is not only error prone, but also a waste of time.10) DO NOT copy a piece of code from the terminal window, paste it somewhere else and try to run it. The format of the copied code would have changed.","@type":"BlogPosting","headline":"NGS Data analysis tips","dateModified":"2017-12-23T16:58:00-05:00","datePublished":"2017-12-23T16:58:00-05:00","@context":"http://schema.org"}</script>
<!-- End Jekyll SEO tag -->
<link rel="stylesheet" href="/assets/main.css"><link type="application/atom+xml" rel="alternate" href="http://localhost:4000/feed.xml" title="Journey of a Bioinformatician" /></head>
<body><header class="site-header" role="banner">

  <div class="wrapper"><a class="site-title" rel="author" href="/">Journey of a Bioinformatician</a><nav class="site-nav">
        <input type="checkbox" id="nav-trigger" class="nav-trigger" />
        <label for="nav-trigger">
          <span class="menu-icon">
            <svg viewBox="0 0 18 15" width="18px" height="15px">
              <path d="M18,1.484c0,0.82-0.665,1.484-1.484,1.484H1.484C0.665,2.969,0,2.304,0,1.484l0,0C0,0.665,0.665,0,1.484,0 h15.032C17.335,0,18,0.665,18,1.484L18,1.484z M18,7.516C18,8.335,17.335,9,16.516,9H1.484C0.665,9,0,8.335,0,7.516l0,0 c0-0.82,0.665-1.484,1.484-1.484h15.032C17.335,6.031,18,6.696,18,7.516L18,7.516z M18,13.516C18,14.335,17.335,15,16.516,15H1.484 C0.665,15,0,14.335,0,13.516l0,0c0-0.82,0.665-1.483,1.484-1.483h15.032C17.335,12.031,18,12.695,18,13.516L18,13.516z"/>
            </svg>
          </span>
        </label>

        <div class="trigger"><a class="page-link" href="/about/">About</a></div>
      </nav></div>
</header>
<main class="page-content" aria-label="Content">
      <div class="wrapper">
        <article class="post h-entry" itemscope itemtype="http://schema.org/BlogPosting">

  <header class="post-header">
    <h1 class="post-title p-name" itemprop="name headline">NGS Data analysis tips</h1>
    <p class="post-meta">
      <time class="dt-published" datetime="2017-12-23T16:58:00-05:00" itemprop="datePublished">Dec 23, 2017
      </time>• <span itemprop="author" itemscope itemtype="http://schema.org/Person"><span class="p-author h-card" itemprop="name">Aarthi Ramakrishnan</span></span></p>
  </header>

  <div class="post-content e-content" itemprop="articleBody">
    <div dir="ltr" style="text-align: left;" trbidi="on">Following are some suggestions while analyzing NGS datasets -<br /><br />1) NEVER delete the outputs of a particular step if the output generation has failed for SOME of the samples. Complete running the pipeline only for the samples which have failed. It might be tempting to delete everything and start all over again (for that particular step), but NEVER do that! It is a waste of your time and resource. Value what you already have! You never know, the next new set of runs might get stuck on queue on the cluster environment. You can always process the remaining samples later on, as managing a small number of samples is easier than managing a large number of samples.<br /><br />2) Before submitting multiple jobs on a cluster, submit a job for one sample first. Also, submit by setting a small walltime. If that works, submit the remaining jobs.<br /><br />3) On a cluster environment, leaving a job running is better than killing a job. Killing a job penalizes your account, and your subsequent jobs are accepted with delays.<br /><br />4) Whenever you cannot get a bioinformatics tool working, try the following:<br />- Read the manual/command line options and check if you have missed something <br />- Check for spelling mistakes when you manually edit files. AVOID MANUALLY EDITING FILES as much as possible in the first place!<br />- Check the file format of your input files. For example, format your BED files to have 6 fields instead of 3<br />- Check if 'chr1' is present in one input file and '1' is present in the other <br />- Email the authors/google the query<br />- Look at the tool's source code <br /><br />5) To check whether a bam file is sorted, if you are able to index the file using samtools index {file name}, then the file is sorted. The headers of the sam file are unreliable. The older versions of samtools report SO:unsorted even though the file could be actually sorted!<br /><br />6) Fold change and log fold change are 2 different terms. Do not mistake one for the other.<br /><br />7) Merging fastq files is not a good idea. Instead, merging the bam files is better.<br /><br />8) Whenever you would like to visualize expression data as a heatmap, it is best to log transform the data before visualization. Vst or rlog transform can be used.<br /><br />9) Automate your process wherever possible! Accomplishing tasks manually is not only error prone, but also a waste of time.<br /><br />10) DO NOT copy a piece of code from the terminal window, paste it somewhere else and try to run it. The format of the copied code would have changed.<br /><br /><br /></div>
  </div><a class="u-url" href="/2017/12/23/ngs-data-analysis-tips.html" hidden></a>
</article>

      </div>
    </main><footer class="site-footer h-card">
  <data class="u-url" href="/"></data>

  <div class="wrapper">

    <h2 class="footer-heading">Journey of a Bioinformatician</h2>

    <div class="footer-col-wrapper">
      <div class="footer-col footer-col-1">
        <ul class="contact-list">
          <li class="p-name">Journey of a Bioinformatician</li><li><a class="u-email" href="mailto:aarthir239@gmail.com">aarthir239@gmail.com</a></li></ul>
      </div>

      <div class="footer-col footer-col-2"><ul class="social-media-list"><li><a href="https://github.com/aartrama"><svg class="svg-icon"><use xlink:href="/assets/minima-social-icons.svg#github"></use></svg> <span class="username">aartrama</span></a></li></ul>
</div>

      <div class="footer-col footer-col-3">
        <p>Aarthi Ramakrishnan&#39;s blog</p>
      </div>
    </div>

  </div>

</footer>
</body>

</html>
